### PR TITLE
unecessary 'required' property in Solution openapi removed

### PR DESCRIPTION
**Pull request by vcarluer**

### CODE REVIEW

File: scenariorun/src/main/kotlin/com/cosmotech/scenariorun/ContainerFactory.kt [hunk 1]
```diff
@@ -1248,8 +1250,10 @@ internal class ContainerFactory(
         }
       }
 
-  private fun getImageName(registry: String, repository: String, version: String? = null): String {
-    val repoVersion = if (version == null) repository else "${repository}:${version}"
+  private fun getImageName(registry: String, repository: String?, version: String? = null): String {
+    if (repository == null) throw IllegalStateException("Solution repository is not defined")
+    val repo = repository ?: "ERROR"
```
Comment (jreynard-code) [inline scenariorun/src/main/kotlin/com/cosmotech/scenariorun/ContainerFactory.kt:1255]: if `repository` is null, you throws an error, `repo` cannot have the value `ERROR`

File: scenariorun/src/main/kotlin/com/cosmotech/scenariorun/dataset/DatasetUtils.kt [hunk 2]
```diff
@@ -32,10 +32,12 @@ internal fun findDatasetsAndConnectors(
     addDatasetAndConnector(
         datasetService, connectorService, organizationId, datasetId, datasets, connectors)
   }
-  val parameterGroupIds = runTemplate.parameterGroups
+  val parameterGroupIds = runTemplate.parameterGroups ?: listOf()
```
Comment (jreynard-code) [inline scenariorun/src/main/kotlin/com/cosmotech/scenariorun/dataset/DatasetUtils.kt:35]: `runTemplate?.parameterGroups` should be enough here

**Review by jreynard-code — CHANGES_REQUESTED** [review]
Some changes to test.
GG ;)

File: scenariorun/src/main/kotlin/com/cosmotech/scenariorun/ContainerFactory.kt (hunk 1 above)
Comment (vcarluer) [reply scenariorun/src/main/kotlin/com/cosmotech/scenariorun/ContainerFactory.kt:1255]: It must be set for Kotlin in order to handle concurrent access but I may find a way to have a better code without this var set.

File: scenariorun/src/main/kotlin/com/cosmotech/scenariorun/dataset/DatasetUtils.kt (hunk 2 above)
Comment (vcarluer) [reply scenariorun/src/main/kotlin/com/cosmotech/scenariorun/dataset/DatasetUtils.kt:35]: runTemplate cannot be null.

File: scenariorun/src/main/kotlin/com/cosmotech/scenariorun/dataset/DatasetUtils.kt [hunk 3]
```diff
@@ -32,10 +32,12 @@ internal fun findDatasetsAndConnectors(
...
   if (parameterGroupIds != null) {
     val parametersIds =
-        (solution.parameterGroups?.filter { it.id in parameterGroupIds }?.map { it.parameters })
+        (solution.parameterGroups?.filter { it.id in parameterGroupIds }?.map {
+              it.parameters ?: listOf()
```
[the remark this answers is not in the document]
Comment (vcarluer) [reply scenariorun/src/main/kotlin/com/cosmotech/scenariorun/dataset/DatasetUtils.kt:39]: yes

File: solution/src/main/kotlin/com/cosmotech/solution/azure/SolutionServiceImpl.kt [hunk 4]
```diff
@@ -153,7 +154,7 @@ internal class SolutionServiceImpl(
       runTemplateId: String
   ) {
     val existingSolution = findSolutionById(organizationId, solutionId)
-    if (!existingSolution.runTemplates.removeIf { it.id == runTemplateId }) {
+    if (!(existingSolution.runTemplates?.removeIf { it.id == runTemplateId } ?: true)) {
```
[the remark this answers is not in the document]
Comment (vcarluer) [reply solution/src/main/kotlin/com/cosmotech/solution/azure/SolutionServiceImpl.kt:157]: runTemplates can be null

File: solution/src/main/kotlin/com/cosmotech/solution/azure/SolutionServiceImpl.kt [hunk 5]
```diff
@@ -308,7 +313,7 @@ internal class SolutionServiceImpl(
       runTemplateId: String,
   ): Solution {
     val solution = findSolutionById(organizationId, solutionId)
-    val validRunTemplateIds = solution.runTemplates.map { it.id }.toSet()
+    val validRunTemplateIds = solution.runTemplates?.map { it.id }?.toSet() ?: setOf()
```
[the remark this answers is not in the document]
Comment (vcarluer) [reply solution/src/main/kotlin/com/cosmotech/solution/azure/SolutionServiceImpl.kt:316]: + || isEmpty()

**Pull request merged**